src/components/CheckboxInput/CheckboxInput.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useEffect, useState } from "react";
+import { useFormContext } from "react-hook-form";
+import check from "../../images/check.svg";
+import { FormValues } from "../Form/Form";
+
+interface CheckboxInputProps {
+  name: keyof FormValues;
+  label: string;
+  error?: { message: string };
+}
+
+const CheckboxInput: React.FC<CheckboxInputProps> = ({ name, label }) => {
+  const {
+    register,
+    formState: { errors, isSubmitted },
+    setValue,
+    getValues,
+  } = useFormContext<FormValues>();
+
+  const [isClicked, setIsClicked] = useState(getValues(name) || false);
+
+  const handleIsClicked = () => {
+    const newValue = !isClicked;
+    setIsClicked(newValue);
+    setValue(name, newValue);
+    console.log("Checkbox state updated:", newValue);
+  };
+
+  useEffect(() => {
+    setIsClicked(getValues(name)); // Sync the checkbox state with the form values
+  }, [getValues, name, isSubmitted]);
+
+  useEffect(() => {
+    if (isSubmitted) {
+      setIsClicked(false); // Reset checkbox state after submission
+    }
+  }, [isSubmitted]);
+
+  const hasError = errors[name]?.message;
+
+  return (
+    <div className="flex flex-col items-center">
+      <div className="flex items-start gap-2.5 ">
+        <div
+          className="relative cursor-pointer min-h-[27px] h-[27px] min-w-[27px]"
+          onClick={handleIsClicked}
+        >
+          <input
+            type="checkbox"
+            {...register(name, {
```

Comment:
Can we make `required` a prop that's passed in, in addition to a `requiredError` string? That'll make this component more flexible and modular for different purposes.